### PR TITLE
upgrade eventgrid  to 2021-12-01; azurerm_eventgrid_domain support for   local_auth_enabled, auto_create_topic_with_first_subscription, auto_delete_topic_with_last_subscription

### PR DESCRIPTION
**Pull request by ms-zhenhua**
https://github.com/hashicorp/terraform-provider-azurerm/issues/14496
image

### CODE REVIEW

File: internal/services/eventgrid/client/client.go [hunk 1]
```diff
@@ -1,21 +1,22 @@
 package client
 
 import (
+	eventgridAlias "github.com/Azure/azure-sdk-for-go/services/eventgrid/mgmt/2021-12-01/eventgrid"
 	"github.com/Azure/azure-sdk-for-go/services/preview/eventgrid/mgmt/2020-10-15-preview/eventgrid"
```
Comment (katbyte) [inline internal/services/eventgrid/client/client.go:5]: how come we are adding a new SDK version here instead of updating the entire service package?

Comment (ms-zhenhua) [reply internal/services/eventgrid/client/client.go:5]: Hi @katbyte , thank you for reviewing my code. The reason is in the [latest 2021-12-01 version](https://docs.microsoft.com/en-us/rest/api/eventgrid/version2021-12-01/event-subscriptions/get#eventsubscriptionfilter) of Event Grid, some of the advancedFilter choices(StringNotBeginsWithAdvancedFilter,StringNotContainsAdvancedFilter,StringNotEndsWithAdvancedFilter) have been removed compared to the [previous version](https://docs.microsoft.com/en-us/rest/api/eventgrid/version2021-06-01-preview/event-subscriptions/get#eventsubscriptionfilter).  To keep users who have already used these filters not affected, I have to temporarily keep resources which use these filter choices with the old API version. I am confirming with the Azure service team how to resolve this issue. When I find a solution, I will resolve it by a new PR. Thanks.

Comment (katbyte) [reply internal/services/eventgrid/client/client.go:5]: @ms-zhenhua - we do not like to mix SDK versions so lets wait to see what the service team says about the missing properties & for a new SDK that fixes this

Comment (tombuildsstuff) [reply internal/services/eventgrid/client/client.go:5]: @ms-zhenhua any update from the service team here?

Comment (ms-zhenhua) [reply internal/services/eventgrid/client/client.go:5]: @tombuildsstuff The service team said they will fix this. I will update this PR after they provide me with the fixed SDK version.

Comment (ms-zhenhua) [reply internal/services/eventgrid/client/client.go]: Hi @katbyte, I have updated the newly released SDK. Please kindly help review it again. Thanks.

**Review by katbyte — APPROVED** [review]
Thanks @ms-zhenhua - LGTM 🏗️

**Pull request merged**